routes/user.rb

```diff
@@ -8,7 +8,8 @@ class Api
       r.post do
         # POST '/api/user/login'
         r.is 'login' do
-          halt(400, 'Could not find user') unless (user = User.by_email(r['email']))
+          halt(400,
+               'Could not find user') unless (user = User.by_email(r['email']) || user = User.first(name: r['email']))
```

Comment:
you don't need to the user =

Reply:
instead change by_email to also search for the username, 

so in models/user.rb adjust that method

it's important to use the lower case functions